azurerm/internal/services/network/vpn_site_resource.go

```diff
@@ -123,7 +123,7 @@ func resourceArmVpnSite() *schema.Resource {
 									"asn": {
 										Type:         schema.TypeInt,
 										Required:     true,
-										ValidateFunc: validation.IntAtLeast(1),
+										ValidateFunc: validation.IntBetween(1, 4294967295),
```

Comment:
@magodo - This upper boundary will break the 32-bit builds, I suspect that this change will need to be reverted.  Is this field expecting asplain for the value, or can it take asdot+ also?